src/System.Private.CoreLib/shared/System/Threading/Semaphore.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32;
+using Microsoft.Win32.SafeHandles;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace System.Threading
+{
+    public sealed partial class Semaphore : WaitHandle
+    {
+        // creates a nameless semaphore object
+        // Win32 only takes maximum count of Int32.MaxValue
+        public Semaphore(int initialCount, int maximumCount) : this(initialCount, maximumCount, null) { }
+
+        public Semaphore(int initialCount, int maximumCount, string name)
+        {
+            VerifyCounts(initialCount, maximumCount);
+
+            bool createdNew;
+            CreateSemaphoreCore(initialCount, maximumCount, name, out createdNew);
+        }
```

Comment:
Rather than refactoring the argument validation into its own method, you could instead just have it in the other ctor and make this constructor be:
```C#
public Semaphore(int initialCount, int maximumCount, string name) : this(initialCount, maximumCount, name, out bool createdNew)
{
}
```
That will require using C# 7.3, though. @jkotas, are we using 7.3 in coreclr yet?

Reply:
Actually, it can be done with 7.2 (which we definitely allow) with just:
```C#
public Semaphore(int initialCount, int maximumCount, string name) :
    this(initialCount, maximumCount, name, out _)
{
}
```